velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -188,21 +183,21 @@ void HiveDataSink::ensurePartitionWriters() {
 
 void HiveDataSink::appendWriter(
     const std::optional<std::string>& partitionName) {
-  // TODO: Wire up serde properties to writer configs.
-  facebook::velox::dwrf::WriterOptions options;
-  options.config = std::make_shared<WriterConfig>();
-  options.schema = inputType_;
   // Without explicitly setting flush policy, the default memory based flush
   // policy is used.
   auto writerParameters = getWriterParameters(partitionName);
   const auto writePath = fs::path(writerParameters.writeDirectory()) /
       writerParameters.writeFileName();
-
-  auto sink = dwio::common::DataSink::create(writePath);
-  writers_.push_back(std::make_unique<Writer>(
-      options, std::move(sink), *connectorQueryCtx_->connectorMemoryPool()));
   writerInfo_.push_back(
       std::make_shared<HiveWriterInfo>(std::move(writerParameters)));
+  auto sink = dwio::common::DataSink::create(writePath);
+
+  auto writerFactory =
+      dwio::common::getWriterFactory(insertTableHandle_->tableStorageFormat());
```

Comment:
```
s/tableStorageFormat/fileFormat/?
```

Reply:
I believe the protocol has a provision for partitionStorageFormat also. So `tableStorageFormat` simply follows the protocol.
https://github.com/prestodb/presto/blob/e825806e6b02e601cc635100514a9442581c05ed/presto-native-execution/presto_cpp/presto_protocol/presto_protocol.h#L2645